### DISCUSSION TITLE
What is the use of "-client" secret?

### DISCUSSION THREAD

**Discussion by joaogbcravo** [jodevsa/wireguard-operator#77] [Q&A]
Maybe I'm missing something, but I don't see why the wireguard-operator is creating a "<VPN>-client" secret. 

As far as I see, it is not mounted by any pod, or used as an environment variable.

The creation code of it is https://github.com/jodevsa/wireguard-operator/blob/e9f36614de6b3f1bac5acb2655d48882ea40fd2c/pkg/controllers/wireguard_controller.go#L698

Can you clarify, please?

**Comment by jodevsa** [comment]
Hi @joaogbcravo ,


The "-client" secret contains the private key for the peer. This needs to be saved so that you can retreive the peer secret using


`kubectl get wireguardpeer peer1 --template={{.status.config}} | bash`


if you try to inspect the bash script generated using 
`kubectl get wireguardpeer peer1 --template={{.status.config}}`

You'll notice that it reads that secret.

**Comment by jodevsa** [comment]
Hi @joaogbcravo,

I was wrong. I somehow thought we where talking about the *-peer secrets 🤦 




Your right. That secret is being created but never gets used 😬 good catch!

**Comment by jodevsa** [comment]
Interesting :D I've added this long time ago to try to generate a peer secret and somehow forgot to remove it 
https://github.com/jodevsa/wireguard-operator/commit/e8e3952d75bf101705ef77f6152d506b0243d1e7

**Comment by jodevsa** [comment]
Are you interested in opening a PR and removing that logic? please feel free to do so :)

**Comment by jodevsa** [comment]
fixed
